Public/Src/Tools/Xldb.Analyzer/XldbAnalyzer.cs

```diff
@@ -0,0 +1,314 @@
+{
+    /// <summary>
+    /// Xldb Analyzers that use the rocksDb instanceand the Xldb API to analyze BXL logs
+    /// </summary>
+    public class Program
+    {
+
+        private const string s_eventStatsAnalyzer = "eventstats";
+        private const string s_dumpPipAnalyzer = "dumppip";
+
+        private Dictionary<string, string> m_commandLineOptions = new Dictionary<string, string>();
+
+        public static int Main(string[] args)
+        {
+            var p = new Program();
+
+            if (!p.ProcessArguments(args))
+            {
+                Console.WriteLine("Invalid arguments passed in, exiting analyzer ...");
+                return 1;
+            }
+
+            if (p.m_commandLineOptions.TryGetValue("/m", out var mode))
+            {
+                if (mode == s_eventStatsAnalyzer)
+                {
+                    return p.AnalyzeEventStats();
+                }
+                else if (mode == s_dumpPipAnalyzer)
+                {
+                    return p.AnalyzeDumpPip();
+                }
+                else
+                {
+                    Console.WriteLine("Invalid mode passed in.");
+                }
+            }
+            else
+            {
+                Console.WriteLine("The mode flag, /m:, is required.");
+            }
+
+            Console.Write($"Valid modes are: /m:{s_eventStatsAnalyzer}, /m:{s_dumpPipAnalyzer}");
+
+            return 1;
+        }
+
+        /// <summary>
+        /// Processes the arguments, checking for the input db path param and the help param
+        /// </summary>
+        public bool ProcessArguments(string[] args)
+        {
+            try
+            {
+                m_commandLineOptions = args.Select(s => s.Split(new[] { ':' }, 2)).ToDictionary(s => s[0].ToLower(), s => s[1]);
+            }
+            catch (IndexOutOfRangeException)
+            {
+                Console.WriteLine("One or more options passed in do not have a key/value. Ie /key:value.");
+                return false;
+            }
+
+            if (!m_commandLineOptions.ContainsKey("/i"))
+            {
+                Console.WriteLine("Input db path required.");
+                return false;
+            }
+
+            if (m_commandLineOptions.ContainsKey("/h"))
+            {
+                Console.WriteLine("Welcome to the xldb analyzer where you can use a rocksdb instance and the Xldb api to analyze BXL logs.");
+                Console.WriteLine($"The current valid analyzers are: /m:{s_eventStatsAnalyzer}, /m:{s_dumpPipAnalyzer}");
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Parses the long representation of the semistable hash from the string passed in (eg. PipC623BCE303738C69 -> 14277462926896108000)
+        /// </summary>
+        public bool ParseSemistableHash(string pipHash, out long parsedHash)
+        {
+            var adjustedOption = pipHash.ToUpper().Replace("PIP", "");

```

Comment:
I recall that someone wrote a helper for this. Let me check...

Reply:
Yes!!! `Pip.TryParseSemiStableHash`

Now I realize this might be problematic for dependencies, but perhaps a helper for other analyzers would be usefull.

---
In reply to: [315940683](https://github.com/microsoft/BuildXL/pull/775#discussion_r315940683) [](ancestors = 315940683)

Reply:
The reason i don't have Pip.TryParseSemiStableHash is because I don't want to be dependent on any more Bxl code than the minimal set that I currently am using. 

---
In reply to: [315941026](https://github.com/microsoft/BuildXL/pull/775#discussion_r315941026) [](ancestors = 315941026,315940683)